src/compiler/transformers/declarations.ts

```diff
@@ -715,16 +731,80 @@ export function transformDeclarations(context: TransformationContext) {
                     shouldPrintWithInitializer(elem) ? elem.initializer : undefined,
                 );
             }
+
             return factory.updateBindingElement(
                 elem,
                 elem.dotDotDotToken,
                 elem.propertyName,
-                filterBindingPatternInitializersAndRenamings(elem.name),
+                filterBindingPatternInitializers(elem.name),
                 shouldPrintWithInitializer(elem) ? elem.initializer : undefined,
             );
         }
     }
 
+    function ensureBindingAliasesInParameterList<T extends any[], N extends Node, P extends number>(original: Node, updater: (...p: T) => N, parametersIndex: P, args: T & Record<P, NodeArray<ParameterDeclaration>>) {

```

Comment:
I'm not sure I like this signature; I'm not aware of this pattern existing in our code and it seems easy to mess up. Can this not just be a switch case on the input node type, or a function you only call on the parameter list, etc?

Reply:
Fixed